bftengine/tests/bcstatetransfer/bcstatetransfer_tests.cpp

```diff
@@ -630,6 +661,39 @@ TEST_F(BcStTest, dstFullStateTransfer) {
   ASSERT_EQ(BCStateTran::FetchingState::NotFetching, stateTransfer_->getFetchingState());
 }
 
+// TBD BC-14432
+TEST_F(BcStTest, sendPrePrepareMsgsDuringStateTransfer) {
+  bftEngine::ReservedPagesClientBase::setReservedPages(&res_pages_mock_);
+  std::once_flag once_flag;
+  ASSERT_NO_FATAL_FAILURE(SendCheckpointSummaries());
+  mockedSrc_->ReplyAskForCheckpointSummariesMsg();
+  auto ss = GetSourceSelector();
+  while (true) {
+    ASSERT_NO_FATAL_FAILURE(
+        AssertFetchBlocksMsgSent(testParams_.expectedFirstRequiredBlockNum, testParams_.expectedLastRequiredBlockNum));
+    mockedSrc_->ReplyFetchBlocksMsg();
+    if (testParams_.expectedLastRequiredBlockNum <= config_.maxNumberOfChunksInBatch) break;
+    testParams_.expectedLastRequiredBlockNum -= config_.maxNumberOfChunksInBatch;
+    this_thread::sleep_for(chrono::milliseconds(20));
+    onTimerImp();
+
+    std::call_once(once_flag, [&] {
+      // For 7 intervals of itemData msgs we need to send 10 PrePrepare so that source gets changed.
+      auto msg = msg_generator_->GeneratePrePrepareMsg(ss.currentReplica());
+      for (uint16_t i = 1; i <= 10; i++) {
+        auto cmsg = make_shared<ConsensusMsg>(msg->type(), msg->senderId());
+        stateTransfer_->handoffConsensusMessage(cmsg);
+      }
+    });
+  }
+  ASSERT_NO_FATAL_FAILURE(AssertFetchResPagesMsgSent());
+  bool doneSending = false;
+  while (!doneSending) mockedSrc_->ReplyResPagesMsg(doneSending);
+  // validate completion
+  ASSERT_TRUE(replica_.onTransferringCompleteCalled_);
+  ASSERT_EQ(BCStateTran::FetchingState::NotFetching, stateTransfer_->getFetchingState());
```

Comment:
It would be good to call here getActualSources() and check that we indeed have 2 sources + after we have the metrics, check all are 0 and the one due to primary is 1.

Reply:
Thanks. Addressed.